src/components/NewCommentForm.tsx

```diff
@@ -41,28 +128,35 @@
 
         <div className="control has-icons-left has-icons-right">
           <input
-            type="text"
+            value={authorEmail}
+            onChange={handleAuthorEmailChange}
+            type="email"
             name="email"
             id="comment-author-email"
             placeholder="email@test.com"
-            className="input is-danger"
+            className={classNames('input', {
+              'is-danger': authorEmailError.length > 0,
+            })}
           />
 
           <span className="icon is-small is-left">
             <i className="fas fa-envelope" />
           </span>
 
           <span
-            className="icon is-small is-right has-text-danger"
+            className={classNames('icon is-small is-right ', {
+              'has-text-danger': authorEmailError.length > 0,
+            })}
             data-cy="ErrorIcon"
           >
             <i className="fas fa-exclamation-triangle" />
           </span>
         </div>
-
-        <p className="help is-danger" data-cy="ErrorMessage">
-          Email is required
-        </p>
+        {authorEmailError && (
+          <p className="help is-danger" data-cy="ErrorMessage">
+            Email is required
+          </p>
```

Comment:
Instead of hardcoding the error message 'Email is required', consider dynamically displaying the `authorEmailError` message. This will make the error handling more flexible and informative.